arwen/contexts/output.go

```diff
@@ -199,6 +199,16 @@ func (context *outputContext) Transfer(destination []byte, sender []byte, gasLim
 	senderAcc, _ := context.GetOutputAccount(sender)
 	destAcc, _ := context.GetOutputAccount(destination)
 
+	payable, err := context.host.Blockchain().IsPayable(destination)
+	if err != nil {
+		return err
+	}
+
+	hasValue := value.Cmp(big.NewInt(0)) == 1
+	if !payable && hasValue {
+		return arwen.ErrAccountNotPayable
+	}
+
```

Comment:
Move these lines above `senderAcc := ...`, to return earlier in case of error.

Reply:
done